### ISSUE TITLE
Error while running run-librivox.sh script

### ISSUE THREAD

**Issue by shahinkl** [mozilla/DeepSpeech#531]
I STARTING Optimization
E label SparseTensor is not valid: indices[246] = [7,90] is out of bounds: need 0 <= index < [12,90]
E 	 [[Node: tower_0/CTCLoss = CTCLoss[ctc_merge_repeated=true, preprocess_collapse_repeated=false, _device="/job:localhost/replica:0/task:0/cpu:0"](tower_0/Reshape_7, tower_0/ToInt64, tower_0/Gather, tower_0/Gather_DequeueMany:1)]]
E 
E Caused by op 'tower_0/CTCLoss', defined at:
E   File "DeepSpeech.py", line 1672, in <module>
E     tf.app.run()
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 44, in run
E     _sys.exit(main(_sys.argv[:1] + flags_passthrough))
E   File "DeepSpeech.py", line 1634, in main
E     train()
E   File "DeepSpeech.py", line 1412, in train
E     results_tuple, gradients, mean_edit_distance, loss = get_tower_results(switchable_data_set, optimizer)
E   File "DeepSpeech.py", line 595, in get_tower_results
E     no_dropout if optimizer is None else dropout_rates)
E   File "DeepSpeech.py", line 481, in calculate_mean_edit_distance_and_loss
E     total_loss = tf.nn.ctc_loss(labels=batch_y, inputs=logits, sequence_length=batch_seq_len)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/ops/ctc_ops.py", line 145, in ctc_loss
E     ctc_merge_repeated=ctc_merge_repeated)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/ops/gen_ctc_ops.py", line 164, in _ctc_loss
E     name=name)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 763, in apply_op
E     op_def=op_def)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2395, in create_op
E     original_op=self._default_original_op, op_def=op_def)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1264, in __init__
E     self._traceback = _extract_stack()
E 
E InvalidArgumentError (see above for traceback): label SparseTensor is not valid: indices[246] = [7,90] is out of bounds: need 0 <= index < [12,90]
E 	 [[Node: tower_0/CTCLoss = CTCLoss[ctc_merge_repeated=true, preprocess_collapse_repeated=false, _device="/job:localhost/replica:0/task:0/cpu:0"](tower_0/Reshape_7, tower_0/ToInt64, tower_0/Gather, tower_0/Gather_DequeueMany:1)]]

Any idea why I am I getting this issue?

**Comment by kdavis-mozilla** [comment]
@shahinkl Are you running using the latest master?

The line numbers in your log don't match with those of `DeepSpeech.py`.

**Comment by shahinkl** [comment]
Yes. I am running with the latest master.
The line number difference is because of slight formatting by the IDE.

**Comment by kdavis-mozilla** [comment]
@shahinkl Were there any problems when you ran `bin/import_librivox.py`? I'm wondering if the librivox data is corrupted.

**Comment by shahinkl** [comment]
I haven't touched that part. For sake of easiness please find the error logs with correct line numbers.

E label SparseTensor is not valid: indices[159] = [4,90] is out of bounds: need 0 <= index < [12,90]
E 	 [[Node: tower_0/CTCLoss = CTCLoss[ctc_merge_repeated=true, preprocess_collapse_repeated=false, _device="/job:localhost/replica:0/task:0/cpu:0"](tower_0/Reshape_7, tower_0/ToInt64, tower_0/Gather, tower_0/Gather_DequeueMany:1)]]
E 
E Caused by op 'tower_0/CTCLoss', defined at:
E   File "/Users/skonadath/PycharmProjects/asr/DeepSpeech.py", line 1628, in <module>
E     tf.app.run()
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 44, in run
E     _sys.exit(main(_sys.argv[:1] + flags_passthrough))
E   File "/Users/skonadath/PycharmProjects/asr/DeepSpeech.py", line 1591, in main
E     train()
E   File "/Users/skonadath/PycharmProjects/asr/DeepSpeech.py", line 1369, in train
E     results_tuple, gradients, mean_edit_distance, loss = get_tower_results(switchable_data_set, optimizer)
E   File "/Users/skonadath/PycharmProjects/asr/DeepSpeech.py", line 569, in get_tower_results
E     calculate_mean_edit_distance_and_loss(batch_set, no_dropout if optimizer is None else dropout_rates)
E   File "/Users/skonadath/PycharmProjects/asr/DeepSpeech.py", line 456, in calculate_mean_edit_distance_and_loss
E     total_loss = tf.nn.ctc_loss(labels=batch_y, inputs=logits, sequence_length=batch_seq_len)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/ops/ctc_ops.py", line 145, in ctc_loss
E     ctc_merge_repeated=ctc_merge_repeated)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/ops/gen_ctc_ops.py", line 164, in _ctc_loss
E     name=name)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 763, in apply_op
E     op_def=op_def)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2395, in create_op
E     original_op=self._default_original_op, op_def=op_def)
E   File "/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1264, in __init__
E     self._traceback = _extract_stack()
E 
E InvalidArgumentError (see above for traceback): label SparseTensor is not valid: indices[159] = [4,90] is out of bounds: need 0 <= index < [12,90]
E 	 [[Node: tower_0/CTCLoss = CTCLoss[ctc_merge_repeated=true, preprocess_collapse_repeated=false, _device="/job:localhost/replica:0/task:0/cpu:0"](tower_0/Reshape_7, tower_0/ToInt64, tower_0/Gather, tower_0/Gather_DequeueMany:1)]]
E

**Comment by kdavis-mozilla** [comment]
What do you mean by:

> I haven't touched that part.

You did run `bin/import_librivox.py` correct?

**Comment by shahinkl** [comment]
I meant that I haven't changed anything there.
I did run bin/import_librivox.py to collect the dataset.

**Comment by shahinkl** [comment]
Let me also do a fresh re-run to check the data inconsistency.

**Comment by kdavis-mozilla** [comment]
Before you try that try removing old checkpoints
```bash
$rm ~/.local/share/deepspeech
```

**Comment by shahinkl** [comment]
Looks like a data issue. Working perfectly in the re-run.
Thanks a lot for your help.

**Comment by kdavis-mozilla** [comment]
Welcome!

**Comment by tilmankamp** [comment]
Constantly getting the same exception for an 8 GPU setup on the Fisher data set.
Seems to be a general issue of the CTCLoss.
The Fisher set constantly fails at the same point in time (around global step 135).
So far I tried to get around it (without success) by:
 - going single node
 - different batch sizes
 - different stddevs
 - different dropoputs
 - offsetting the training samples - higher offset samples (which are longer) lead to slightly earlier failing
 - moving form 8 kHz samples to 16 kHz ones (re-sampled the whole data set)

**Comment by THUHJ** [comment]
@tilmankamp  Hi, have you fixed this problem?
I met the same err on WSJ datasets and librispeech set, when I run the code in ds2-v2 branch.

(Btw, I can run the code on Deepspeech master correctly. So I guess it is not a problem relevant to data set. )

**Comment by tilmankamp** [comment]
This kind of error typically occurs, if the number of audio frames of a sample is lower than the number of  characters of its transcription. In most cases this is due to wrong sample transcriptions. However: At least once there was also some edge case with a very fast speaker. For the Fisher data set I identified this problematic samples and either excluded them or corrected their transcriptions. We closed this bug, as this procedure is known to fix this kind of problems.
Certain network geometry settings can influence sensitivity to this effect. That could be the reason for the difference to ds2-v2. 
@reuben Any input on this?

**Comment by reuben** [comment]
I think it's unlikely that DS2 has a big influence here, the convolution there is equivalent to the striding we do in `audio.py` (in terms of the reduction in time steps). You should check if there's no corrupt transcription or audio files in the disk. I've seen cases where there's tons of corrupt audio files (0 frames) and the end result is the same: too many characters for too few audio frames.
